spyder/plugins/editor/widgets/codeeditor.py

```diff
@@ -4063,18 +4063,16 @@ def _handle_completions(self):
             cursor.select(QTextCursor.WordUnderCursor)
             text = to_text_string(cursor.selectedText())
 
-        if (len(text) >= self.automatic_completions_after_chars
-                and self._last_key_pressed_text):
-            self.document_did_change(text)

```

Comment:
Did you read the `git blame` for these lines? I mean, I'd like to know who did them and what was the rationale behind them to understand why they were added in the first place.

Reply:
@goanpeca added this lines to the function

Reply:
>        if (len(text) >= self.automatic_completions_after_chars
>                and self._last_key_pressed_text):

That logic is/was there to ensure the minimum chars to trigger a completionit worked properly . It seems that now that is broken with these changes?

Reply:
I think the `document_did_change` statement needs to go outside this if, but perhaps we need to preserve it, as @goanpeca mentioned.

@steff456, what do you think?

Reply:
`document_did_change` should be called always, so it should be outside this if

Reply:
I agree. Then I think the solution is to leave the current `if` and move `document_did_change` before it.

Reply:
If I put the `if`, then there's some cases of the backspace that won't be called because the last instruction is to generate the completions

Reply:
Ok, that is a corner case that needs to be addressed @steff456 without deleting code that is actually accomplishing a feature.

Reply:
Ok, but if we don't leave the if we'll break showing completions after a certain number of characters.

So please try to fix the case when the number of chars is 0. If that's not possible, we'd have to drop that case from this PR.